File: 646 (Maximum Length of Pair Chain)/javascript/Solution.js
/**
 * @param {number[][]} pairs
 * @return {number}
 */
var findLongestChain = function(pairs) {
  pairs.sort((a,b) => a[1]-b[1]);

  let res =0;
  let max = Number.MIN_SAFE_INTEGER;

  for(const pair of pairs){
      if (pair[0]>max){
          res++;
          max = pair[1];
      }
  }
  return res;
};

File: 646 (Maximum Length of Pair Chain)/python/Solution.py
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        pairs.sort(key = lambda x: x[1])

        res = 0
        max = float('-inf')

        for a,b in pairs:
            if a>max:
                res+=1
                max = b
        
        return res